### DISCUSSION TITLE
How to use it? Where do I find the documentation?

### DISCUSSION THREAD

**Discussion by JackHewson** [OWASP/www-project-csrfguard#212] [Q&A] [answered]
I want to use CSRFGuard 4.3 in my java web application, but I do not find any documentation how to integrate it.
The only source I've found was https://medium.com/@pandey.ritwik/implementing-csrf-protection-in-java-based-application-using-owasp-1d62466654da, but this seems outdated.

**How to integrate/implement CSFRGuard 4.3?**

**Comment by forgedhallpass** [comment, marked as answer]
Hello @JackHewson,

Please check out the bundled [csrfguard-test-jsp](https://github.com/OWASP/www-project-csrfguard/tree/master/csrfguard-test/csrfguard-test-jsp) application which covers most of the features offered by CSRFGuard. 

On a high level the following steps would be needed:
* Add the right maven dependencies (see the [README](https://github.com/OWASP/www-project-csrfguard/blob/master/csrfguard/src/main/resources/csrfguard.properties) page).
* Choose your desired settings in the [configuration](https://github.com/OWASP/www-project-csrfguard/blob/master/csrfguard/src/main/resources/csrfguard.properties) file.
* Update your [web.xml](https://github.com/OWASP/www-project-csrfguard/blob/master/csrfguard-test/csrfguard-test-jsp/src/main/webapp/WEB-INF/web.xml).
* Reference the [`csrfguard.js`](https://github.com/OWASP/www-project-csrfguard/blob/master/csrfguard/src/main/resources/csrfguard.js) file from your HTML files (this is not needed if you are using JSPs).

P.S. feel free to contribute the documentation in a PR once you've managed to set it up.
